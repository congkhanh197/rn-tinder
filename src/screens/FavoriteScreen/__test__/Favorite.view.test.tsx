import React from 'react';
import Favorite, {IProps} from '../Favorite.view';
import renderer from 'react-test-renderer';
describe('Test Favorite view', () => {
  const props: IProps = {
    peopleList: [
      {
        dob: 'testDob',
        location: {
          street: 'test',
        },
        md5: 'testMd5',
        name: {
          first: 'mr',
          last: 'test',
          title: 'name',
        },
        password: 'test',
        phone: '0124',
        picture: 'test',
      },
    ],
  };
  test('should render', () => {
    const tree = renderer.create(<Favorite {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
