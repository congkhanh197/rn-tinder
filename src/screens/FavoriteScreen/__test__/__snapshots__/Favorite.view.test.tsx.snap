// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Favorite view should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgray",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "dob": "testDob",
          "location": Object {
            "street": "test",
          },
          "md5": "testMd5",
          "name": Object {
            "first": "mr",
            "last": "test",
            "title": "name",
          },
          "password": "test",
          "phone": "0124",
          "picture": "test",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 4,
              "marginHorizontal": 75,
              "marginTop": 20,
              "width": 600,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#f9f9f9",
                "borderBottomColor": "gray",
                "borderBottomWidth": 1,
                "height": 100,
                "width": 598,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "gray",
                "borderRadius": 55,
                "borderWidth": 1,
                "height": 110,
                "justifyContent": "center",
                "position": "absolute",
                "top": 25,
                "width": 110,
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 50,
                "height": 100,
                "justifyContent": "center",
                "overflow": "hidden",
                "position": "absolute",
                "top": 30,
                "width": 100,
                "zIndex": 2,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://i1.wp.com/www.mvhsoracle.com/wp-content/uploads/2018/08/default-avatar.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 50,
                  "height": 100,
                  "width": 100,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 50,
                "height": 100,
                "justifyContent": "center",
                "overflow": "hidden",
                "position": "absolute",
                "top": 30,
                "width": 100,
                "zIndex": 3,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "test",
                }
              }
              style={
                Object {
                  "borderRadius": 50,
                  "height": 100,
                  "width": 100,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginBottom": 30,
                "marginTop": 60,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "gray",
                  "fontSize": 20,
                }
              }
            >
              My name is
            </Text>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 25,
                }
              }
            >
              Name. Mr Test
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "marginBottom": 10,
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderTopColor": "green",
                  "borderTopWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 35,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "green",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderTopColor": "transparent",
                  "borderTopWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 35,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderTopColor": "transparent",
                  "borderTopWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 35,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderTopColor": "transparent",
                  "borderTopWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 35,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderTopColor": "transparent",
                  "borderTopWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 35,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
